src/components/MapView/Legends/index.tsx

```diff
@@ -138,8 +232,11 @@ export interface LegendsProps extends WithStyles<typeof styles> {
 interface LegendItemProps
   extends WithStyles<typeof styles>,
     PropsWithChildren<{}> {
```

Comment:
I would personally refactor this to `extends Pick<LayerType, 'id' | 'title' ... > `. This won't work too well however if the optionals are different between the layer type and props type. What do you think?

Reply:
I tried to refactor based on your suggestion, but got into trouble trying to display the Analysis map layer. It has no `id` and `type` so I need to keep the optionals here. Once the AnalysisLayer is added to LayerType, I think it can be done